sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -215,3 +215,56 @@ case class GpuSize(child: Expression, legacySizeOfNull: Boolean)
     }
   }
 }
+
+case class GpuSortArray(base: Expression, ascendingOrder: Expression)
+    extends GpuBinaryExpression with ExpectsInputTypes {
+
+  override def left: Expression = base
+
+  override def right: Expression = ascendingOrder
+
+  override def dataType: DataType = base.dataType
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(ArrayType, BooleanType)
+
+  override def checkInputDataTypes(): TypeCheckResult = base.dataType match {
+    case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
+      ascendingOrder match {
+        // replace Literal with GpuLiteral here
+        case GpuLiteral(_: Boolean, BooleanType) =>
+          TypeCheckResult.TypeCheckSuccess
+        case _ =>
```

Comment:
for easier diagnostics let us log what was provided. Same on L244

Reply:
Done.